planner/planner.go

```diff
@@ -349,6 +349,20 @@ func (p *MyPlanner) homePageHandler(ctx *gin.Context) {
 	ctx.Redirect(http.StatusMovedPermanently, "/v1/")
 }
 
+func (p *MyPlanner) getOptimalPlan(ctx *gin.Context) {
+	req := &PlanningReq{}
+
+	if err := ctx.ShouldBindJSON(req); err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err})
+	}
+
+	if plan, err := p.Solver.SolveHungarianOptimal(ctx, req); err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err})
+	} else {
+		ctx.JSON(http.StatusOK, gin.H{"plan": plan})
```

Comment:
Are we just returning the new plan as text on html page for the moment?

Reply:
I am making a new commit so that we return more info including time slot, place name and category.